File: sdk-android/src/main/java/cu/suitetecsa/sdk/android/utils/StringUtils.java
package cu.suitetecsa.sdk.android.utils;

import androidx.annotation.NonNull;

import org.jetbrains.annotations.NotNull;

import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.Date;
import java.util.Locale;

public class StringUtils {
    public static long toSeconds(@NotNull String time) {
        String[] parts = time.split(":");
        long totalSeconds = 0;

        for (String part : parts) {
            int SECONDS_PER_MINUTE = 60;
            totalSeconds = totalSeconds * SECONDS_PER_MINUTE + Long.parseLong(part);
        }

        return totalSeconds;
    }

    public static long toDateMillis(String date) throws ParseException {
        Calendar calendar = Calendar.getInstance();
        calendar.setTime(toDate(date));
        return calendar.getTimeInMillis();
    }

    public static Date toDate(String date) throws ParseException {
        return new SimpleDateFormat("dd-MM-yy", Locale.getDefault()).parse(date);
    }

    public static long toBytes(@NotNull String data) {
        String count = data.replaceAll("[GMKBT]", "");
        String unit = data.split(" ")[data.split(" ").length - 1].toUpperCase();
        return (long) (Double.parseDouble(count) * Math.pow(1024, "BKMGT".indexOf(unit.charAt(0))));
    }
}

File: sdk-android/src/main/java/io/github/suitetecsa/sdk/android/utils/StringUtils.kt
package io.github.suitetecsa.sdk.android.utils

import java.text.ParseException
import java.text.SimpleDateFormat
import java.util.Calendar
import java.util.Date
import java.util.Locale
import kotlin.math.pow

private const val SecondsPerMinute = 60

object StringUtils {
    @JvmStatic
    fun toSeconds(time: String): Long {
        val parts = time.split(":".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
        var totalSeconds: Long = 0
        for (part in parts) {
            totalSeconds = totalSeconds * SecondsPerMinute + part.toLong()
        }
        return totalSeconds
    }

    @JvmStatic
    fun isActive(data: String): Boolean = data != "no activos"

    @JvmStatic
    @Throws(ParseException::class)
    fun toDateMillis(date: String): Long {
        val calendar = Calendar.getInstance()
        toDate(date)?.let { calendar.setTime(it) }
        return calendar.getTimeInMillis()
    }

    @Throws(ParseException::class)
    fun toDate(date: String): Date? =
        date.let { SimpleDateFormat("dd-MM-yy", Locale.getDefault()).parse(it) }

    @JvmStatic
    fun fixDateString(date: String): String = date.replace("-", "/")

    @JvmStatic
    fun toBytes(data: String): Long {
        val count = data.replace("[GMKBT]".toRegex(), "")
        val unit = data.split(" ".toRegex()).dropLastWhile { it.isEmpty() }
            .toTypedArray()[
            data.split(" ".toRegex()).dropLastWhile { it.isEmpty() }
                .toTypedArray().size - 1
        ].uppercase(Locale.getDefault())
        return (count.toDouble() * 1024.0.pow("BKMGT".indexOf(unit[0]).toDouble())).toLong()
    }
}
